### ISSUE TITLE
Add libRocket package

### ISSUE THREAD

**Issue by Dinno** [ruslo/hunter#278]
http://librocket.com/

**Comment by ruslo** [comment]
Sources: https://github.com/libRocket/libRocket
*note*: use CMake as a build tool

**Comment by Dinno** [comment]
I am starting adding this one. It may be tough and lengthy, so I will trace progress here.

**Comment by ruslo** [comment]
Do you need repo in https://github.com/hunter-packages?

**Comment by Dinno** [comment]
I don't know for the moment. I will ask you if I need it.

**Comment by Dinno** [comment]
I don't know howe to solve problem with libRocket's placement of CMakeLists.txt file. It is placed inside Build directory, and ExternalProject_add directive can't find it. I tried to find solution on the Web, but couldn't. Do you know how to solve the problem without patch for libRocket? If not then I need repo for the patch.

**Comment by ruslo** [comment]
> I tried to find solution on the Web, but couldn't. Do you know how to solve the problem without patch for libRocket?

There is no nice solution, you just need to create CMakeLists.txt in top.

> If not then I need repo for the patch.

Created: https://github.com/hunter-packages/libRocket

I think it make sense to send this patch to upstream.

**Comment by Dinno** [comment]
Yes, I plan to send them the patch. And I plan to remove pre-built solutions and projects entirely. It is strange to pre-build something that can be easily built in place. I will write some politically correct version of this consideration in pool request :)

**Comment by Dinno** [comment]
Ruslan, I have a question. Do I understand right that when we build final program modules we add hunter_add_package directive into project's CMakeLists.txt file, but when we create package for library, we add hunter_add_package directives for it's dependencies into package's hunter.cmake file? Or in case when we use patched package we can place the directive into CMakeLists.txt?

**Comment by Dinno** [comment]
After writing the question I decided to make some research :), and found some projects in hunter-packages which contain some hunter_add_package directives in their CMakerLists.txt. So perhaps I answered my question myself - I need to place directives into CMakerLists.txt not into hunter.cmake.

**Comment by ruslo** [comment]
> add hunter_add_package directives for it's dependencies into package's hunter.cmake file?

No, there is no need to set dependency in Hunter. It's one of the main Hunter features. Dependencies specified in package itself. This allow to use optional dependencies, like `BUILD_WITH_OPENSSL=ON` will download OpenSSL and `BUILD_WITH_OPENSSL=OFF` will not.

> I need to place directives into CMakerLists.txt not into hunter.cmake

Yes, exactly.
